Client/Client.Qt/Widgets/ChatHistory.cpp

```diff
@@ -17,37 +17,78 @@ ChatHistory::ChatHistory(QWidget* parent) : QWidget(parent), _messageList()
     connect(_scrollArea.get(), SIGNAL(scrolled()), this, SLOT(resizeVisible()));
 }
 
-void ChatHistory::addMessage(const QString& message, quint64 utc, const QString& user)
+void ChatHistory::addMessage(const QString& message, quint64 utc, ReplyWidget* reply, const QString& user)
```

Comment:
If you can try to to split this method into several methods. It's too big. (If you can't or don't want just ignore my comments)